pkg/resourceinterpreter/configurableinterpreter/configurable.go

```diff
@@ -2,156 +2,326 @@ package configurableinterpreter
+		UpdateFunc: func(_, _ interface{}) { c.updateConfiguration() },
+		DeleteFunc: func(_ interface{}) { c.updateConfiguration() },
+	})
+	return c
 }
 
-// HookEnabled tells if any hook exist for specific resource gvk and operation type.
-func (c *ConfigurableInterpreter) HookEnabled(kind schema.GroupVersionKind, operationType configv1alpha1.InterpreterOperation) bool {
-	if !c.configManager.HasSynced() {
-		klog.Errorf("not yet ready to handle request")
-		return false
-	}
-	accessors, exist := c.configManager.LuaScriptAccessors()[kind]
-	if !exist {
-		return false
-	}
-	switch operationType {
-	case configv1alpha1.InterpreterOperationAggregateStatus:
-		return len(accessors.GetStatusAggregationLuaScript()) > 0
-	case configv1alpha1.InterpreterOperationInterpretHealth:
-		return len(accessors.GetHealthInterpretationLuaScript()) > 0
-	case configv1alpha1.InterpreterOperationInterpretDependency:
-		return len(accessors.GetDependencyInterpretationLuaScript()) > 0
-	case configv1alpha1.InterpreterOperationInterpretReplica:
-		return len(accessors.GetReplicaResourceLuaScript()) > 0
-	case configv1alpha1.InterpreterOperationInterpretStatus:
-		return len(accessors.GetStatusReflectionLuaScript()) > 0
-	case configv1alpha1.InterpreterOperationRetain:
-		return len(accessors.GetRetentionLuaScript()) > 0
-	case configv1alpha1.InterpreterOperationReviseReplica:
-		return len(accessors.GetReplicaRevisionLuaScript()) > 0
-	}
-	return false
+// HookEnabled tells if any hook exist for specific resource type and operation.
+func (c *ConfigurableInterpreter) HookEnabled(objGVK schema.GroupVersionKind, operation configv1alpha1.InterpreterOperation) bool {
+	_, enabled := c.getInterpreter(objGVK, operation)
+	return enabled
 }
 
 // GetReplicas returns the desired replicas of the object as well as the requirements of each replica.
-func (c *ConfigurableInterpreter) GetReplicas(object *unstructured.Unstructured) (int32, *workv1alpha2.ReplicaRequirements, error) {
-	klog.Infof("ConfigurableInterpreter Execute ReviseReplica")
-	customAccessor := c.configManager.LuaScriptAccessors()[object.GroupVersionKind()]
-	if len(customAccessor.GetReplicaResourceLuaScript()) == 0 {
-		return 0, nil, fmt.Errorf("customized interpreter operation GetReplicas for %q not found", object.GroupVersionKind())
+func (c *ConfigurableInterpreter) GetReplicas(object *unstructured.Unstructured) (replicas int32, require *workv1alpha2.ReplicaRequirements, enabled bool, err error) {
+	eng, enabled := c.getInterpreter(object.GroupVersionKind(), configv1alpha1.InterpreterOperationInterpretReplica)
+	if !enabled {
+		return
 	}
 
-	luaScript := customAccessor.GetReplicaResourceLuaScript
-	klog.Infof("lua script %s", luaScript)
+	rets, err := eng.Invoke(object)
+	if err != nil {
+		return
+	}
+
+	replicas, err = rets[0].Int32()
+	if err != nil {
+		return
+	}
 
-	return 0, nil, nil
+	if r := rets[1]; !r.IsNil() {
+		require = &workv1alpha2.ReplicaRequirements{}
+		err = r.Into(require)
```

Comment:
@XiShanYongYe-Chang make sure if script returns nil, here we shall return nil or empty struct?